smart-contracts/assembly/contracts/FT/wrapper.ts

```diff
@@ -59,7 +49,7 @@ export class TokenWrapper {
    * @returns name of the token.
    */
   name(): string {
-    return this._name;
+    return bytesToString(call(this._origin, 'name', NoArg, 0));
```

Comment:
I think we should keep the name in the Class property to avoid making several calls to the contract.
Or maybe keep a cached value of it

Reply:
> to avoid making several calls to the contract

With my solution, the call to `name` is made if and only if it is needed, and if you need to use the value more than once you can indeed cache it by assigning it to a const.
With this one, it is called everytime you create a new TokenWrapper.

But I can understand your point of view and maybe im just over-engineering the gas optimisation.